### ISSUE TITLE
Wrong Options "formatter" type?

### ISSUE THREAD

**Issue by GiedriusGrabauskas** [palantir/tslint#1427]
### Bug Report

- __TSLint version__: `3.13.0`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: Node.js API

`ILinterOptionsRaw` and `ILinterOptions` param `formatter` accept only `string`. But in code, it's possible to use `Function`.

#### Usage sample:
```ts
import * as Lint from "tslint/lib/lint";
import tslint from "gulp-tslint";

// Custom formatter class
export class Formatter extends Lint.Formatters.AbstractFormatter {
    public format(failures: Lint.RuleFailure[]): string {
        var failuresJSON = failures.map((failure: Lint.RuleFailure) => failure.toJson());
        return JSON.stringify(failuresJSON);
    }
}


gulp.task("tslint", () =>
    gulp.src("source.ts")
        .pipe(tslint({
            formatter: Formatter  // Custom formatter
        }))
        .pipe(tslint.report())
);
```
P.S. I using `gulp-tslint`, but param `options` is the same.

#### `tslint` code:
* `lint` function take `formatter` from options: https://github.com/palantir/tslint/blob/master/src/tslint.ts#L123

* Calling function `findFormatter` and return same function:
https://github.com/palantir/tslint/blob/master/src/formatterLoader.ts#L27

* Creating `formatter` instance:
https://github.com/palantir/tslint/blob/master/src/tslint.ts#L125

* But interface accept only `string`:
https://github.com/palantir/tslint/blob/master/src/lint.ts#L51
https://github.com/palantir/tslint/blob/master/src/lint.ts#L58

**Comment by DovydasNavickas** [comment]
I've updated `findFormatter` method definition here:
https://github.com/palantir/tslint/blob/master/src/formatterLoader.ts#L25
PR: https://github.com/palantir/tslint/pull/1443

As @GiedriusGrabauskas stated, this code [here](https://github.com/palantir/tslint/blob/master/src/formatterLoader.ts#L27):
```TypeScript
export function findFormatter(name: string, formattersDirectory?: string) {
    if (typeof name === "function") {
        return name;
    }
    <...>
}
```
accepts the function and returns it immediately instead of looking for formatter by name, but the parameter type is constrained to string only.

It should accept a `Function` in addition to string as so:
```TypeScript
export function findFormatter(name: string | Function, formattersDirectory?: string)
```

For now it was not possible to pass custom formatter function (class) properly (except by casting the class to a `<string>` explicitly, which is a hacky way).

**Comment by JKillian** [comment]
Thanks all, I think this should be fixed by #1443

**Comment by panuhorsmalahti** [comment]
Any idea when 3.15 is going to be released? gulp-tslint is basically waiting for this.

**Comment by adidahiya** [comment]
@panuhorsmalahti we're releasing 3.15 today
